onnx/defs/tensor/defs.cc

```diff
@@ -3943,4 +3943,75 @@ ONNX_OPERATOR_SET_SCHEMA(
           return true;
         }));
 
+static const char* TensorScatter_ver24_doc = R"DOC(
+TensorScatter performs kv cache updates for Attention calculations. The past and present cache tensors have the
+same shape, with the sequence length dimension being max_seqlen, so the sizes of these tensors do not need to 
+grow between iterations. 
+The optional write_indices input indicates the write index for each sample in the batch, assumed to be zero 
+if not provided. During the prefill phase of attention, only the first two inputs are needed. During the decode 
+phase, write_indices is also needed so that the incoming k and v can be appended after the last valid token 
+for each sample in the batch.
+
+In order to perform kv caching in place, as is the common practice for efficient inference, the execution 
```

Comment:
nit: Would be better to not to refer to execution provider, which is an ORT concept, not ONNX. May be say something like
```
This is intended to support an explicit representation of in-place kv cache updates (instead of modelling it via a functional Concat operation).
```